src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/ServiceDescriptionService.java

```diff
@@ -772,7 +774,7 @@ public ServiceDescriptionType updateOpenApi3ServiceDescription(Long id, String u
      * @param ignoreWarnings
      * @param serviceDescription
      * @throws OpenApiParser.ParsingException if there are errors in the openapi3 description document
-     * @throws UnhandledWarningsException     if ignoreWarnings is false and parser returns warnings from openapi
+     * @throws UnhandledWarningsException if ignoreWarnings is false and parser returns warnings from openapi
      */
     private void parseOpenApi3ToServiceDescription(String url, String serviceCode, boolean ignoreWarnings,
```

Comment:
parseOpenApi3ToServiceDescription is really quite complex, it would be really great to make it simpler somehow. Could it be split into smaller parts, etc? Now it feels like a risky part that could lead to some bugs.

The fact that you can mix generated and non-generated endpoints makes it kind of unavoidable that the aggregate implementation will be complex, unfortunately.

It would have been good to add a test for the failing case (before fixing it)